Performance and cleanup

jgerits opened the pull request:
I did some performance tweaking of the package the results are:


Trino Tests:
| **Test Name**   | **Min(ms)** | **Max(ms)** | **Mean(ms)** | **Rounds** |
| --------------- | ----------- | ----------- | ------------ | ---------- |
| Inserts Single  | 444         | 712         | 489,9        | 100        |
| Inserts Complex | 3.616       | 6.528       | 4.393,9      | 100        |
| Read Single     | 900         | 1.177       | 963,7        | 100        |
| Read Complex    | 61.886      | 69.018      | 63.799,7     | 10         |
| Update Single   | 258         | 530         | 306,7        | 100        |
| Update Complex  | 510         | 754         | 580,6        | 100        |
| Delete Single   | 260         | 378         | 306,6        | 100        |
| Multi Delete    | 255         | 745         | 316,2        | 100        |

Changed package tests:

| **Test Name**   | **Min(ms)** | **Max(ms)** | **Mean(ms)** | **Rounds** |
| --------------- | ----------- | ----------- | ------------ | ---------- |
| Inserts Single  | 325         | 632         | 349,09       | 100        |
| Inserts Complex | 3414        | 5398        | 4161,92      | 100        |
| Read Single     | 586         | 1257        | 672,06       | 100        |
| Read Complex    | 32693       | 42644       | 36268,4      | 10         |
| Update Single   | 136         | 214         | 146,48       | 100        |
| Update Complex  | 374         | 608         | 452,92       | 100        |
| Delete Single   | 134         | 271         | 146,46       | 100        |
| Multi Delete    | 135         | 250         | 144,55       | 100        |

jgerits committed:
cleanup code

jgerits committed:
cleanup and optimize code

mosabua commented:
Could you consider contributing to the new official trino client? 

https://github.com/trinodb/trino-csharp-client

Find us on Trino slack to chat

Copilot committed:
Initial plan

Copilot committed:
Setup NuGet package creation and auto-release workflow

- Updated TrinoClient.csproj with proper package metadata (PackageId: JGerits.TrinoClient, version 1.0.0)
- Replaced existing workflow with comprehensive CI/CD pipeline
- Added build-and-test job with artifact upload
- Added auto-release job with smart checks for existing releases and NuGet packages

Co-authored-by: jgerits <6541449+jgerits@users.noreply.github.com>

Copilot committed:
Improve workflow error handling based on code review

- Add stderr redirect for git diff to handle first commit case more gracefully
- Add logging for NuGet API status codes
- Add error suppression for grep to handle missing index.json

Co-authored-by: jgerits <6541449+jgerits@users.noreply.github.com>

Copilot committed:
Exclude test project changes from triggering package releases

- Removed TrinoClientTests pattern from RELEVANT_PATTERNS
- Test changes will no longer trigger new NuGet package releases
- TrinoClientTests project already exists as a starter for unit tests

Co-authored-by: jgerits <6541449+jgerits@users.noreply.github.com>

jgerits committed:
Add automated NuGet packaging and GitHub releases

Copilot committed:
Initial plan

Copilot committed:
Fix compiler warnings - remove unused fields and fix obsolete property usage

Co-authored-by: jgerits <6541449+jgerits@users.noreply.github.com>

Copilot committed:
Implement IDisposable to properly dispose HttpClient resources

Co-authored-by: jgerits <6541449+jgerits@users.noreply.github.com>

Copilot committed:
Fix Dispose pattern based on code review feedback

Co-authored-by: jgerits <6541449+jgerits@users.noreply.github.com>

Copilot committed:
Add explicit handler disposal since HttpClient doesn't own them

Co-authored-by: jgerits <6541449+jgerits@users.noreply.github.com>

Copilot committed:
Update disposal comment for clarity

Co-authored-by: jgerits <6541449+jgerits@users.noreply.github.com>

jgerits committed:
Fix compiler warnings and implement IDisposable for HTTP resource management

Copilot committed:
Initial plan

Copilot committed:
Add unit test project and exclude integration tests from CI

Co-authored-by: jgerits <6541449+jgerits@users.noreply.github.com>

Copilot committed:
Add comprehensive unit tests and testing documentation

Co-authored-by: jgerits <6541449+jgerits@users.noreply.github.com>

Copilot committed:
Exclude NuGet package creation from pull request builds

Co-authored-by: jgerits <6541449+jgerits@users.noreply.github.com>

jgerits committed:
Separate unit tests from integration tests requiring Trino

Copilot committed:
Initial plan

Copilot committed:
Add comprehensive unit tests for DataSize, ErrorLocation, QueryError, ExecuteQueryV2Request, ConnectorId, and TransactionId classes

Co-authored-by: jgerits <6541449+jgerits@users.noreply.github.com>

Copilot committed:
Add unit tests for HostAddress, QueryId, and ErrorCode classes

Co-authored-by: jgerits <6541449+jgerits@users.noreply.github.com>

Copilot committed:
Fix test edge cases based on code review feedback

Co-authored-by: jgerits <6541449+jgerits@users.noreply.github.com>

jgerits committed:
Merge pull request #5 from jgerits/copilot/add-extra-unit-tests

Add unit tests for core model classes

Copilot committed:
Performance optimizations: reduce allocations and eliminate redundant operations

Copilot committed:
Add HTTP compression, async optimizations, and enhanced retry logic

Copilot committed:
Add SocketsHttpHandler with connection pooling, HTTP/2, and keep-alive optimizations

Copilot committed:
Add Entity Framework Core provider for Trino
